solr/core/src/java/org/apache/solr/core/ConfigSetService.java

```diff
@@ -353,86 +323,55 @@ protected NamedList loadConfigSetFlags(CoreDescriptor cd, SolrResourceLoader loa
   public abstract void deleteConfig(String configName) throws IOException;
 
   /**
-   * Delete a file in config
+   * Delete files in config
    *
-   * @param configName the config to delete
-   * @param fileName the file to delete
-   * @throws IOException if an I/O error occurs
-   */
-  public abstract void deleteFileFromConfig(String configName, String fileName) throws IOException;
-
-  /**
-   * Set config metadata
-   *
-   * @param configName the config
-   * @param data metadata to be set in config
-   * @throws IOException if an I/O error occurs
-   */
-  public abstract void setConfigMetadata(String configName, Map<Object, Object> data) throws IOException;
-
-  /**
-   * Set config metadata
-   *
-   * @param configName the config
-   * @param data metadata to be set in config
+   * @param filesToDelete a list of file paths to delete
    * @throws IOException if an I/O error occurs
    */
-  public abstract void setConfigMetadata(String configName, byte[] data) throws IOException;
+  public abstract void deleteFilesFromConfig(List<String> filesToDelete) throws IOException;
 
   /**
-   * Update config metadata
-   *
-   * @param configName the config
-   * @param data metadata to be updated in config
-   * @throws IOException if an I/O error occurs or if config does not exist
-   */
-  public abstract void updateConfigMetadata(String configName, Map<Object, Object> data) throws IOException;
-
-  /**
-   * Update config metadata
-   *
-   * @param configName the config
-   * @param data metadata to be updated in config
+   * Set the config metadata
+   * If config does not exist, it will be created and set metadata on it
+   * Else metadata will be updated
+   * @param configName the config name
+   * @param data the metadata to be set on config
    * @throws IOException if an I/O error occurs
    */
-  public abstract void updateConfigMetadata(String configName, byte[] data) throws IOException;
+  public abstract void setConfigMetadata(String configName, Map<String, Object> data) throws IOException;
 
   /**
-   * Get config metadata
+   * Get the config metadata
    *
-   * @param configName the config
+   * @param configName the config name
    * @return the config metadata
```

Comment:
Does it return null?  It'd be nice on callers if we can guarantee non-null.  Also mention wether the result is mutable.